*** autodoc.py.orig	2011-02-25 17:34:09.000000000 -0500
--- autodoc.py	2011-04-04 13:39:17.000000000 -0400
***************
*** 553,561 ****
              # if isattr is True, the member is documented as an attribute
              isattr = False
  
!             if want_all and membername.startswith('_'):
!                 # ignore members whose name starts with _ by default
                  skip = True
              elif (namespace, membername) in attr_docs:
                  # keep documented attributes
                  skip = False
--- 553,563 ----
              # if isattr is True, the member is documented as an attribute
              isattr = False
  
!             if want_all and membername.startswith('__init__'):
                  skip = True
+             elif want_all and membername.startswith('_'):
+                 # ignore members whose name starts with _ by default
+                 skip = not self.options.private_members
              elif (namespace, membername) in attr_docs:
                  # keep documented attributes
                  skip = False
***************
*** 727,732 ****
--- 729,735 ----
          'show-inheritance': bool_option, 'synopsis': identity,
          'platform': identity, 'deprecated': bool_option,
          'member-order': identity, 'exclude-members': members_set_option,
+         'private-members': bool_option,
      }
  
      @classmethod
***************
*** 880,886 ****
          'members': members_option, 'undoc-members': bool_option,
          'noindex': bool_option, 'inherited-members': bool_option,
          'show-inheritance': bool_option, 'member-order': identity,
!         'exclude-members': members_set_option,
      }
  
      @classmethod
--- 883,889 ----
          'members': members_option, 'undoc-members': bool_option,
          'noindex': bool_option, 'inherited-members': bool_option,
          'show-inheritance': bool_option, 'member-order': identity,
!         'exclude-members': members_set_option, 'private-members': bool_option,
      }
  
      @classmethod
***************
*** 1135,1141 ****
  
      # flags that can be given in autodoc_default_flags
      _default_flags = set(['members', 'undoc-members', 'inherited-members',
!                           'show-inheritance'])
  
      # standard docutils directive settings
      has_content = True
--- 1138,1144 ----
  
      # flags that can be given in autodoc_default_flags
      _default_flags = set(['members', 'undoc-members', 'inherited-members',
!                           'show-inheritance', 'private-members'])
  
      # standard docutils directive settings
      has_content = True
